internal/controller/ols_app_server_assets.go

```diff
@@ -279,21 +278,29 @@ func (r *OLSConfigReconciler) generateOLSConfigMap(ctx context.Context, cr *olsv
 		olsConfig.QueryFilters = queryFilters
 	}
 
-	userDataCollectorConfig := UserDataCollectorConfig{
-		DataStorage: "/app-root/ols-user-data",
-		LogLevel:    cr.Spec.OLSDataCollectorConfig.LogLevel,
+	appSrvConfigFile := AppSrvConfigFile{
+		LLMProviders: providerConfigs,
+		OLSConfig:    olsConfig,
 	}
-	var appSrvConfigFile AppSrvConfigFile
 	if dataCollectorEnabled {
-		appSrvConfigFile = AppSrvConfigFile{
-			LLMProviders:            providerConfigs,
-			OLSConfig:               olsConfig,
-			UserDataCollectorConfig: userDataCollectorConfig,
+		appSrvConfigFile.UserDataCollectorConfig = UserDataCollectorConfig{
+			DataStorage: "/app-root/ols-user-data",
+			LogLevel:    cr.Spec.OLSDataCollectorConfig.LogLevel,
 		}
-	} else {
-		appSrvConfigFile = AppSrvConfigFile{
-			LLMProviders: providerConfigs,
-			OLSConfig:    olsConfig,
+	}
+
+	if cr.Spec.OLSConfig.IntrospectionEnabled {
```

Comment:
@xrajesh I didn't remove the cr.Spec.OLSConfig.IntrospectionEnabled flag, it's just the new code reacts to it here. With MCP in the OLS service, ols_config.IntrospectionEnabled is gone and is replaced with the mcp_servers config section, instead of the 1:1 mapping of the IntrospectionEnabled flags.